transaction/src/manifest/lexer.rs

```diff
@@ -364,13 +359,59 @@ impl Lexer {
     }
 
     fn unexpected_char(&self) -> LexerError {
-        LexerError::UnexpectedChar(self.text[self.current.full_index - 1], self.current)
+        let mut end = self.previous;
+        end.full_index += 1;
+
+        LexerError {
+            error_kind: LexerErrorKind::UnexpectedChar(self.text[self.current.full_index - 1]),
+            span: Span {
+                start: self.current,
+                end,
+            },
+        }
     }
+
+    fn unexpected_char_previous(&self) -> LexerError {
+        // If advance() is used, we want to get the position of previous token not the current one
+        let mut end = self.previous;
+        end.full_index += 1;
+
+        LexerError {
+            error_kind: LexerErrorKind::UnexpectedChar(self.text[self.previous.full_index]),
+            span: Span {
+                start: self.previous,
+                end,
+            },
+        }
+    }
+}
+
+pub fn lexer_error_diagnostics(s: &str, err: LexerError) -> String {
+    let (title, label) = match err.error_kind {
+        LexerErrorKind::UnexpectedEof => (
+            "unexpected end of file".to_string(),
+            "end of file".to_string(),
```

Comment:
Perhaps this should be `unexpected end of file` as well?

Reply:
Done